src/display/text_layer.js

```diff
@@ -65,6 +65,9 @@ const ascentCache = new Map();
 let _canvasContext = null;
 
 function getCtx(lang = null) {
+  if (_canvasContext && _canvasContext.canvas.lang !== (lang || "")) {
+    cleanupTextLayer();
```

Comment:
Unfortunately this could break things quite badly outside of the viewer, since when the library API is used directly multiple PDF documents could be active at once and this could "pull the rug" on another currently rendering textLayer from another document (and thus with another `lang` parameter).

---

Hence I believe that the "correct" solution would be to instead change `_canvasContext` to a `Map` and thus cache (potentially multiple) temporary canvases on the `lang` value.